CSharpMath/Structures/Result.cs

```diff
@@ -62,23 +67,19 @@ namespace CSharpMath.Structures {
     public static implicit operator Result<T>(ResultImplicitError error) =>
       new Result<T>(error.Error);
     public Result Bind(Action<T> method) {
-      if (method is null) throw new ArgumentNullException(nameof(method));
```

Comment:
Why this was removed? (or why it was removed only here) 
I think it is better to have the same checks for all similar methods&types.
I.e. if `SpanResult.Bind<TResult>(Func<TResult> method)` has the check then I would expect that `Result<T>.Bind<TResult>(Func<T, Result> method)` has the same.

Reply:
My preference is removing the null check in `SpanResult.Bind<TResult>(Func<TResult> method)`. As the `Bind` methods are used heavily, these checks make extra debugging steps. These seem unnecessary given that we have nullable reference types and the method will instantly throw anyways.